### PR TITLE
feat: add .showPrefixes property

### LINKED ISSUE #66: Custom prefixes joiner
Hi, this is your fellow happy user from the [wix/Detox](https://wix.github.io/Detox) project. 👋 

**Do you mind if we add a sort of custom function to join prefixes?**

Currently your logic is hardcoded like that: `prefixes => [${prefixes.join(',')]` – I'd like to be able to customize this convention to produce logs like here below:

image

P.S. To illustrate the current logic:

image

### PR DESCRIPTION
**Pull request by noomorph**
Resolves #66.

- [x] #66

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -40,16 +40,18 @@ The most basic usage involves just creating a Bunyan logger which writes raw obj
...
-    streams: [{
-        level:  'info',
-        type:   'raw',
-        stream: bunyanDebugStream.create({
-            basepath: __dirname, // this should be the root folder of your project.
-            forceColor: true
-        })
-    }],
-    serializers: bunyanDebugStream.serializers
+  name: 'myLog',
```
Comment (noomorph) [inline README.md:43]: Just making sure – this reformatting was automatic:

image

Comment (jwalton) [reply README.md:43]: Apparently when I added prettier, I ran it on `src` but not on anything else.  :P

File: src/BunyanDebugStream.ts [hunk 2]
```diff
@@ -349,7 +357,14 @@ class BunyanDebugStream extends Writable {
             }
         }
 
-        const joinedPrefixes = prefixes.length > 0 ? `[${prefixes.join(',')}] ` : '';
+        let joinedPrefixes = '';
+        if (this._showPrefixes && prefixes.length > 0) {
+            if (typeof this._showPrefixes === 'function') {
+                joinedPrefixes = `${this._showPrefixes(prefixes)} `;
```
Comment (jwalton) [inline src/BunyanDebugStream.ts:363]: If `_showPrefixes()` throws an error here, we may want to fall back on

```ts
joinedPrefixes = `[${prefixes.join(',')}] `;
```

?

**Review by jwalton — CHANGES_REQUESTED** [review]

File: src/BunyanDebugStream.ts (hunk 2 above)
Comment (noomorph) [reply src/BunyanDebugStream.ts:363]: This is not in line with your date (showDate) formatting behavior. It has no similar fallbacks AFAIC. I'd rather implement this concern separately.

Comment (jwalton) [reply src/BunyanDebugStream.ts:363]: Fair enough!  :)

**Pull request merged**